### ISSUE TITLE
poll: CKEditor. Should we change kama theme to moono-lisa-TS?

### ISSUE THREAD

**Issue by mahotilo** [Typesetter/Typesetter#377]
I have tried several CKEditor themes and I am not happy with any. But, if made a step with elFinder, should we move to Moono-Lisa in CKEditor?
It looks more like... contemporary. Monochrome, flat...

This will bring some alignment problems! You've been warned:)

![image](https://user-images.githubusercontent.com/14929385/71075189-52696b80-218c-11ea-9895-9f2bbe41933f.png)
![image](https://user-images.githubusercontent.com/14929385/71075207-585f4c80-218c-11ea-823e-5ac3577d6b46.png)
![image](https://user-images.githubusercontent.com/14929385/71075229-61e8b480-218c-11ea-9a64-71582c7a839d.png)

**Comment by mahotilo** [comment]
I customized theme a little. Now it looks better

![image](https://user-images.githubusercontent.com/14929385/71082958-ac712d80-219a-11ea-9f5b-55376882ef3e.png)
![image](https://user-images.githubusercontent.com/14929385/71082999-b98e1c80-219a-11ea-9d66-78892b4a7a77.png)
![image](https://user-images.githubusercontent.com/14929385/71083023-c4e14800-219a-11ea-84fa-6ff54709f7a5.png)

**Comment by mahotilo** [comment]
Maybe add a little bluish?
![image](https://user-images.githubusercontent.com/14929385/71235722-a51a6300-2305-11ea-8420-657345ce91f3.png)

And I have solved the problem of dialog resizing due to higher inputs.
I began to like this moono-lisa-TS theme:)

**Comment by mahotilo** [comment]
And now with scaled image preview

![image](https://user-images.githubusercontent.com/14929385/71310194-f0be3100-2419-11ea-9013-376761fb3032.png)

![image](https://user-images.githubusercontent.com/14929385/71310161-90c78a80-2419-11ea-89fc-25893885ae05.png)

**Comment by juek** [comment]
That's also a hard decision. AFAIK the topic is well-researched and the facts are speaking slightly **against** monochromatic icons for applications. They are easier to read at first, but more difficult to distinguish later. Similar discussions in different communities often led to colored symbols coming back later again (e.g. CorelDRAW or currently Blender).

From an aesthetic point of view, I give a clear yes. 
Modernization-pressure is strong.

Yep, let's go for Moono-Lisa. :)

**Comment by mahotilo** [comment]
I have just pulled a request
https://github.com/Typesetter/Typesetter/pull/388

and add a colour variant for your estimation

![image](https://user-images.githubusercontent.com/14929385/71319108-784f8280-24a2-11ea-8252-32707ea3fc81.png)

**Comment by juek** [comment]
I just pushed moono-lisa

![grafik](https://user-images.githubusercontent.com/704283/71319166-f7d95380-249a-11ea-9043-ca58e47efa12.png)

**Comment by juek** [comment]
we **could** make it configurable

**Comment by mahotilo** [comment]
It will be reasonable decision, but theming is not typical for our solid style CMS.

**Comment by mahotilo** [comment]
What do you think of my skin.js in moono-lisa-TS-c.
It has two components.
1. Same height for all tabs in dialogs
2. Advanced image preview (fit image to preview box and some simulation of sizing and floating)

It seems to me that at least "Same height" and "Fit image to preview box" can be implemented in TS

**Comment by juek** [comment]
Seems as if it doesn't load here. I'm afraid it must be part of the CKEditor build. How did you manage to load it?

**Comment by mahotilo** [comment]
> How did you manage to load it?

Just changed skin.js from /moono-lisa/

Maybe problem is due to hardcoded theme name
`...CKEDITOR.skin.name="moono-lisa-TS";`

**Comment by juek** [comment]
No, it's because the skin.js is [part of the CKEditor build](https://github.com/Typesetter/Typesetter/blob/master/include/thirdparty/ckeditor/ckeditor.js#L551)

**Comment by mahotilo** [comment]
It works for me.
- I created dir ckeditor/skins/moono-lisa-TS
- copied files from moono-lisa including skin.js
- changed skin.js
- and it runs without any tricks

**Comment by juek** [comment]
Actually, it appears that the skin name passed via config must deviate from the built-in theme in order to load the external script.js

![grafik](https://user-images.githubusercontent.com/704283/71647634-3cb28500-2cfa-11ea-910f-65ef5e8b7b3c.png)

The marked part wants to be considered as text :)

**Comment by mahotilo** [comment]
Yep, it will look like this if you indeed  insert picture with such size:)

**Comment by mahotilo** [comment]
> Actually, it appears that the skin name passed via config must deviate from the built-in theme in order to load the external script.js

CKEditor...

**Comment by juek** [comment]
Since Mozilla seems to have dropped native drag image resizing recently (Chrome + Safari never had that) the only way to resize inline images is numerical via dialog fields. 

I wonder if we should add [Enhanced Image](https://ckeditor.com/cke4/addon/image2) to the build.

**Comment by mahotilo** [comment]
Frankly, I would prefer to set the image size with my own hands anyway.

**Comment by juek** [comment]
> Frankly, I would prefer to set the image size with my own hands anyway.

I don't use inline images at all, normally, except in SimpleBlog posts. 
But Typesetter resamples scaled inline images by default unless disabled in Config. Not a bad feature - some folks like it.

**Comment by juek** [comment]
> I wonder if we should add Enhanced Image to the build.

Just tested it again. It's too limited.

**Comment by mahotilo** [comment]
I suppose we should close this issue and the corresponding PR https://github.com/Typesetter/Typesetter/pull/388.

Monochrome Moono-lisa is the default theme for CKEditor 4, so it should be known and familiar to the end user. If you are still thinking about the color icons, it can be done later or as a plugin.

I made the [plugin](https://github.com/mahotilo/CKE_MoonoLisaM) for myself and can change CKEditor theme CSS and JS as I want without intruding into the CMS core:)

What do you think of that?

**Comment by juek** [comment]
In my POV the monochrome icons are clear enough and, surprisingly, I don't miss the colors.

I should probably upload a fresh build because I think there are parts of kama in the current one (my fault). I will close when it's done.

**Comment by mahotilo** [comment]
FYI
I have made the [CKE_Themes plugin ](https://github.com/mahotilo/CKE_Themes) that allows you to change CKEditor theme to one from `https://ckeditor.com/cke4/addons/skins`.
It was not simple because I had to revert some TS styles from inline_edit.css. But I succeeded eventually.

**Comment by mahotilo** [comment]
> I should probably upload a fresh build because I think there are parts of kama in the current one (my fault). I will close when it's done.

Is everything cleared of Kama now? Can we close?

**Comment by juek** [comment]
Now it is

https://github.com/Typesetter/Typesetter/commit/b5ae9dc725710b120cd56540d24ac975e4386044  https://github.com/Typesetter/Typesetter/commit/bdf6609d601fe7f13499c07068e3e2fb814c024a

**Comment by mahotilo** [comment]
Are you sure of deleting styles.js?
The claim

> - **skin.js**: the only JavaScript part of the skin that registers the skin, its browser specific files and its icons and defines the Chameleon feature,

**Comment by juek** [comment]
It's part of the build now

**Comment by juek** [comment]
As I understand, CKEditor Builder will combine everything required for the initial load into ckeditor.js, skin, plugins, etc, to speed up loading. Only some assets and dialog JS will remain stand-alone and loaded on demand.

**Comment by mahotilo** [comment]
Thx, I got it.

And one last question on this topic.
Have you noticed that we have `icons.png` and `icons_hidpi.png` twice. One pair is in `/ckeditor/skins/moono-lisa/` and the other is in `/ckeditor/plugins/`?
I tried to remove one of them, but due to caching failed to understand which one is used for what. But I've noticed that all the paths in page's inline CKE uses only icons from `/ckeditor/plugins/`. And my additional fork theme 'moono-lisa-M' works perfectly without `icons.png` in theme's dir just using one from `/ckeditor/plugins/`.
